### PR TITLE
Encrypted media test google migration: syntax

### PR DESCRIPTION
**Pull request by squapp**
this adds following migrated test cases:

encrypted-media/clearkey-mp4-syntax-mediakeys.html
encrypted-media/clearkey-mp4-syntax-mediakeysession.html
encrypted-media/clearkey-mp4-syntax-mediakeysystemaccess.html

encrypted-media/drm-mp4-syntax-mediakeys.html
encrypted-media/drm-mp4-syntax-mediakeysession.html
encrypted-media/drm-mp4-syntax-mediakeysystemaccess.html

Originally the "syntax" test was only one file. For clarity the tests are now split into 3 files for Clear Key and DRM, respectively.

### CODE REVIEW

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysession.js]: InvalidAccessError should be TypeError for non persistent sessions (and for persistent sessions with invalid sessionIds).

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysession.js]: Ditto

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysession.js]: Ditto

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysession.js]: Ditto

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysession.js]: Ditto

File: encrypted-media/scripts/syntax-mediakeysystemaccess.js
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysystemaccess.js]: NotSupportedError here as well.

File: encrypted-media/scripts/syntax-mediakeysystemaccess.js
Comment (jdsmith3000) [inline encrypted-media/scripts/syntax-mediakeysystemaccess.js]: I believe this is NotSupportedError too.

**Comment by jdsmith3000** [comment]
With error type comments aside, the rest of these tests LGTM.  Please hold merging until after review by AlexeyKor.

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: Update description: initDataType should be supported.

File: encrypted-media/scripts/syntax-mediakeys.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeys.js]: This applies to all "then"/"catch" in these files.

File: encrypted-media/scripts/syntax-mediakeys.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeys.js]: Could you separate all functions with empty lines?

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysystemaccess.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysystemaccess.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysystemaccess.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysystemaccess.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeys.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeys.js]: nit: add ; 
This applies to all promise_tests in these files.

**Review by AlexeyKor — APPROVED** [review]
Looks good to me.

**Comment by jdsmith3000** [comment]
LGTM.  I want to merge this today so that all pending new tests are checked in.  Please comment if there is pending work coming.

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysession.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysession.js]: nit: add ;

File: encrypted-media/scripts/syntax-mediakeysystemaccess.js
Comment (AlexeyKor) [inline encrypted-media/scripts/syntax-mediakeysystemaccess.js]: nit: add ;

**Review by AlexeyKor — COMMENTED** [review]
Could you fix the last 3 nits?

**Comment by jdsmith3000** [comment]
I see there are missing ";" on three lines.  I'm not sure how to pick that up before merging, since the files are in a migration-3 repository at Fraunhofer Fokus.  I could do the merge and then do a separate commit with the fixes.  Okay?

**Comment by mwatson2** [comment]
You can commit the changes to Fraunhofer's branch:

git remote fokus <url for Fraunhofer fork>
git fetch fokus migration-3
git checkout fokus/migration-3

make the changes and commit them

git push fokus HEAD:migration-3

...Mark

**Comment by jdsmith3000** [comment]
I opened this issue:  https://github.com/w3c/web-platform-tests/issues/3912

**Comment by jdsmith3000** [comment]
@mwatson2 I see your guidance.  I'll work that and close the tracking issue.

**Comment by jdsmith3000** [comment]
This is odd.  The last commit triggered lint errors in a file not touched in the commit and not even in the encrypted-media directory:

$ ./lint
PRESENT-TESTHARNESSCSS: web-animations/interfaces/Animation/effect.html Explicit link to testharness.css present
There was 1 error (PRESENT-TESTHARNESSCSS: 1)

**Comment by mwatson2** [comment]
No, that was not your fault. @jgraham merged a bunch of stuff from some
other repo and introduced this problem into master. So the master branch is
failing lint now.

It was a few hours ago, so disappointing it has not been fixed yet.

...Mark

**Comment by jdsmith3000** [comment]
Will this clear when the root problem is fixed, or must we do something to re-check our PR?

**Comment by jdsmith3000** [comment]
I checked in web-animations and there is currently no effect.html file.

**Comment by mwatson2** [comment]
On Thu, Oct 6, 2016 at 1:25 PM, jdsmith3000 <notifications@github.com>
wrote:

> I checked in web-animations and there is currently no effect.html file.
>

​In master there is:
https://github.com/w3c/web-platform-tests/blob/master/web-animations/interfaces/Animation/effect.html
​

**Comment by jdsmith3000** [comment]
It doesn't make sense to me that an unrelated pull request is blocked on a check that fails on a file already in w3c/master...

**Comment by mwatson2** [comment]
On Thu, Oct 6, 2016 at 1:38 PM, jdsmith3000 <notifications@github.com>
wrote:

> It doesn't make sense to me that an unrelated pull request is blocked on a
> check that fails on a file already in w3c/master...
>
​The CI tools run on what the repo would look like if your PR was merged.
So, because the problem is in master, all PR CI runs of lint will fail.

Obviously this can't happen if the only way things are merged into master
is from PRs, since Git won't let you merge if the CI run fails.

​@jgraham must have merged his content some other way. This also happened a
few weeks back.

...Mark

**Pull request merged**
